### DISCUSSION TITLE
loco new app workflow automation

### DISCUSSION THREAD

**Discussion by tjyang** [loco-rs/loco#273] [Ideas]
## Feature Request

Please add more command line options to automate new loco app  interactive creation by loco command line from loco-cli crate.

**Existing interactive workflow**

![image](https://github.com/loco-rs/loco/assets/378638/b36b432a-b983-42e0-b218-ef7f6dacdfa2)

**Describe the solution you'd like**

 loco  --name myapp3 --db=sqlite --starter=saas new

**Comment by kaplanelad** [comment]
Hey @tjyang, I believe that introducing more flags to the CLI can add complexity for both users and in the codebase. Our default and recommended choice is Postgres. However, users always have the flexibility to customize their preferences by making changes in the configuration.

**Comment by tjyang** [comment]
- After some source code grepping, following can achieve what I need for now.
```
# --> LOCO_TEMPLATE= {lightweight-service|rest-api|saas}
LOCO_APP_NAME=myapp3 LOCO_TEMPLATE=saas loco new -p /tmp
```

**Reply by lerina** [reply]
hehehe, perfect. I was looking for something like this. 
Thank you for sharing

**Comment by tjyang** [comment]
Hi 
I am trying the automated new app creation via command line.

- egrep search showing there is example in  generate-starters.trycmd file.
```
[me@rocky9t01a tmp]$ cat /etc/redhat-release
Rocky Linux release 9.5 (Blue Onyx)
[me@rocky9t01a tmp]$ loco -V
loco 0.14.0
[me@rocky9t01a loco]$ egrep  LOCO_TEMPLATE loco-cli/tests/cmd/starters/generate-starters.trycmd
$ ALLOW_IN_GIT_REPO="" LOCO_APP_NAME="test_saas_template" LOCO_FOLDER_NAME=saas_template LOCO_TEMPLATE=saas loco new  --db postgres --assets serverside --bg async
$ ALLOW_IN_GIT_REPO="true" LOCO_APP_NAME="test_rest_api_template" LOCO_FOLDER_NAME=rest_api_template LOCO_TEMPLATE=rest-api loco new --db sqlite --bg async
$ ALLOW_IN_GIT_REPO="true" LOCO_APP_NAME="test_lightweight_template" LOCO_FOLDER_NAME=lightweight_template LOCO_TEMPLATE=lightweight-service loco new
[me@rocky9t01a loco]$

```
- The quick test shows the ALLOW_IN_GIT_REPO and LOCO_APP_NAME variable not taken.
   I still need to enter the value interactively.
```
[me@rocky9t01a tmp]$ ALLOW_IN_GIT_REPO="true" LOCO_APP_NAME="test_saas_template" LOCO_FOLDER_NAME=saas_template LOCO_TEMPLATE=saas loco new  --db postgres --assets serverside --bg async
✔ ❯ You are inside a git repository. Do you wish to continue? · yes
✔ ❯ App name? · myapp
  Loco app generated successfully in:
  /tmp/myapp
   database: You've selected `postgres` as your DB provider (you should have a postgres instance to connect to)
[me@rocky9t01a tmp]$
```

![image](https://github.com/user-attachments/assets/6cf2a6d6-3008-46c6-a965-5ea30ad312a8)

**Comment by kaplanelad** [comment]
We also automate the template creation with shell command., take a look here: https://github.com/loco-rs/loco/blob/master/loco-gen/tests/db.rs#L47

**Comment by tjyang** [comment]
Thank you @kaplanelad. 

Template creation using command arguments is much cleaner. 

```
[me@rocky9t01a ~]$ loco new  -n myapp --db sqlite --bg async --assets  serverside -a

🚂 Loco app generated successfully in:
/home/me/myapp


[me@rocky9t01a ~]$

```
